### ISSUE TITLE
Cannot do chunked reads on compressed files. To read, set blocksize=None

### ISSUE THREAD

**Issue by fredms** [dask/dask#7758]
I tried to read data from compressed file to dask bag as follows:

`bag = db.read_text('[path/to/gzip/file/data.gz]', blocksize="2.5GB").str.strip().str.split('\t')`

but I got error:
Cannot do chunked reads on compressed files. To read, set `blocksize=None`

If I changed blocksize to be None as suggested, the bag only had 1 partition, which was dumped to 1 worker for further computation no matter how many workers were available.

I also tried` bag.repartition(npartitions=6)`, but all works were still assigned to 1 worker (even though there were 2 workers available).

How to read data from gzip file and have multiple partitions which can be assigned to multiple workers for computation?

**Comment by mrocklin** [comment]
Yeah, there isn't a good way to break apart a GZipped file.  It's not a
great compression format for parallel reading.  My recommendation would be
to uncompress the file before using a tool like Dask and then maybe in the
future consider using different compression formats.

**Comment by fredms** [comment]
Thanks @mrocklin. 

My intention is to read data from compressed file(s) directly. Now, my workaround is to have multiple gzip files (one file == one partition).

I am wondering if there is any compression format which supports parallel reading well with dask?

**Comment by GenevieveBuckley** [comment]
You mention partitions, so I assume you're using dataframes. If that's the case, you should probably look at using the Parquet format. 

There's a bit about why it's useful in the last part of the [dask dataframe best practices](https://docs.dask.org/en/latest/dataframe-best-practices.html) (in short, it's a lot easier to do parallel loading of data, and should also be able to handle the compression on disk for you). There's some [examples here](https://examples.dask.org/dataframes/01-data-access.html#Write-to-Parquet).

(For reference, you can also see the full list of file formats with good dask dataframe integration [here](https://docs.dask.org/en/latest/dataframe-create.html), although they won't all necessarily be suitable for you.)

**Comment by fredms** [comment]
Thanks @GenevieveBuckley for the reply.

I am using dask bag to read from compressed files and bag also has partitions.

1. if it reads from ONE gzip file, the bag has 1 partition, which can only be assigned to 1 worker for further computation no matter how many workers available.
2. if it reads from TWO gzip files, the bag has 2 partitions, which can possibly be assigned to 2 workers.

**Comment by GenevieveBuckley** [comment]
I'm not super clear on what type of data you are putting into dask bag. If you're able to clarify what it is you're doing, you might get some better suggestions from other people here.

**Comment by fredms** [comment]
Data has 7 columns with string values
```
col1     col2   col3   col4  col5   col6   col7
"a"      "b"     "c"    "d"   "e"    "f"    "g"
.....
```
The goals:
1. Read data from 1 large compressed file (gzip) to dask dataframe, 
2. Then repartition it to 2 partitions 
3. Assigned them to 2 workers (1 partition per worker) for further computation (inferencing).

For 1 & 2, I don't see any feasible way to read the compressed file(s) to dask dataframe directly. Please let me know if there is a way to achieve this.
My way is to read data to dask bag first and then use `bag.to_dask_dataframe()` and` .repartition(npartitions=2).`

For 3, Even though I explicitly repartition it to 2 partitions, only 1 worker is processing data.

Question:
1. how to do chunked reads so that all workers are in use?
2. any other compressed format supports parallel reading well with dask?

**Comment by mrocklin** [comment]
If you have tabular data then I recommend using a structured format like
Parquet with Dask Dataframe.  If you are restricted to using flat files
then the "big pile of gzipped files" that you proposed earlier is pretty
standard unfortunately :)

**Comment by fredms** [comment]
yes, the gzip file contains tsv format files that cannot be changed.

**Comment by mrocklin** [comment]
Technically you could probably use lzma / xz files.  But that's somewhat
exotic.  Pile-of-gzip for the win.
